manager/snapshot.go

```diff
@@ -1,13 +1,61 @@
 package manager
 
 import (
+	"fmt"
+	bsutil "github.com/longhorn/backupstore/util"
 	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
 )
 
-func (m *VolumeManager) ListSnapshots(volumeName string) (map[string]*longhorn.Snapshot, error) {
+func (m *VolumeManager) ListSnapshotsCR(volumeName string) (map[string]*longhorn.Snapshot, error) {
 	return m.ds.ListVolumeSnapshotsRO(volumeName)
 }
 
-func (m *VolumeManager) GetSnapshot(snapName string) (*longhorn.Snapshot, error) {
+func (m *VolumeManager) GetSnapshotCR(snapName string) (*longhorn.Snapshot, error) {
 	return m.ds.GetSnapshotRO(snapName)
 }
+
+func (m *VolumeManager) DeleteSnapshotCR(snapName string) error {
+	return m.ds.DeleteSnapshot(snapName)
+}
+
+func (m *VolumeManager) CreateSnapshotCR(snapshotName string, labels map[string]string, volumeName string) (*longhorn.Snapshot, error) {
+	if volumeName == "" {
+		return nil, fmt.Errorf("volume name required")
+	}
+
+	for k, v := range labels {
+		if strings.Contains(k, "=") || strings.Contains(v, "=") {
+			return nil, fmt.Errorf("labels cannot contain '='")
+		}
+	}
```

Comment:
nit: I saw some places having this same validation. Can we place in a common pkg to reuse here.

Reply:
Fixed. Thank you